src/modules/Dashboard/features/Managers/Popups/AddManagerPopup/AddManagerPopup.tsx

```diff
@@ -0,0 +1,239 @@
+import { useEffect, useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import { useForm, useWatch } from 'react-hook-form';
+import { yupResolver } from '@hookform/resolvers/yup';
+
+import { EmptyState, Modal, Spinner } from 'shared/components';
+import { StyledErrorText, StyledModalWrapper } from 'shared/styles';
+import { useFormError } from 'modules/Dashboard/hooks';
+import { Roles, VALIDATION_DEBOUNCE_VALUE } from 'shared/consts';
+import { Mixpanel, getErrorMessage, isManagerOrOwner } from 'shared/utils';
+import { Languages, postAppletInvitationApi } from 'api';
+import { useAppDispatch } from 'redux/store';
+import { useAsync } from 'shared/hooks';
+import { banners, users, workspaces } from 'redux/modules';
+
+import { USER_ALREADY_INVITED, defaultValues } from './AddManagerPopup.const';
+import { AddManagerPopupSchema } from './AddManagerPopup.schema';
+import { AddManagerPopupProps, AddManagerFormValues, Fields } from './AddManagerPopup.types';
+import { AddManagerForm } from './AddManagerForm';
+
+export const AddManagerPopup = ({
+  appletId,
+  onClose,
+  popupVisible,
+  workspaceInfo,
+  'data-testid': dataTestid,
+}: AddManagerPopupProps) => {
+  const { t, i18n } = useTranslation('app');
+  const [successBannerId, setSuccessBannerId] = useState<string | null>(null);
+  const rolesData = workspaces.useRolesData();
+  const appletRoles = appletId ? rolesData?.data?.[appletId] : undefined;
+
+  // Workspace Name field is only available if the workspace has no managers yet
+  const isWorkspaceNameVisible = !!workspaceInfo && !workspaceInfo.hasManagers;
+
+  const { ownerId } = workspaces.useData() || {};
+  const respondentsData = users.useAllRespondentsData();
+  const participants = (respondentsData?.result ?? []).map(({ details }) => {
+    const { respondentSecretId, respondentNickname, subjectId } = details[0];
+
+    return {
+      subjectId,
+      secretId: respondentSecretId,
+      nickname: respondentNickname,
+      // TODO: Populate tag
+      // https://mindlogger.atlassian.net/browse/M2-5861
+      tag: null,
+    };
+  });
+
+  const defaults = {
+    ...defaultValues,
+    language: i18n.language as Languages,
+    workspaceName: workspaceInfo?.name,
+  };
+
+  const {
+    control,
+    formState: { isValid, isDirty },
+    handleSubmit,
+    register,
+    reset,
+    setError,
+    setValue,
+    unregister,
+  } = useForm<AddManagerFormValues>({
+    resolver: yupResolver(AddManagerPopupSchema(isWorkspaceNameVisible)),
+    defaultValues: defaults,
+    mode: 'all',
+    delayError: VALIDATION_DEBOUNCE_VALUE,
+  });
```

Comment:
The `delayError` here is causing some errors to not display when changing fields before the delay passes, and you can end up in a form without errors that doesn't let you submit

https://github.com/ChildMindInstitute/mindlogger-admin/assets/2916156/e8bd2a84-3317-4930-a9a2-18d5c1e3abd9


![CleanShot 2024-04-30 at 11 11 31](https://github.com/ChildMindInstitute/mindlogger-admin/assets/2916156/e1eeb4a0-3a8e-4d61-8a01-17fb41f67aaf)

Reply:
Good catch! I was able to reproduce this, too. Fixed it and maintained good UX by removing the error delay and running validation `onBlur` instead. Applied same fix to the other two forms I've authored that used the same strategy.